### ISSUE TITLE
Remix app Admin API version change 

### ISSUE THREAD

**Issue by johnark88** [Shopify/shopify-app-template-remix#376]
# Issue summary
I am creating an app to modify `CheckoutBranding` on `Shopify+ Store`. I created an app using the Shopify CLI
I have a `app/models/branding.server.ts`, which takes `admin.graphql` and this query 
```grapql
 mutation checkoutBrandingUpsert($checkoutBrandingInput: CheckoutBrandingInput!, $checkoutProfileId: ID!) {
  checkoutBrandingUpsert(checkoutBrandingInput: $checkoutBrandingInput, checkoutProfileId: $checkoutProfileId) {
    checkoutBranding {
      designSystem {
        # This property group applies to global corner radius token values
        cornerRadius {
          small 
          ............
``` 
Query can be found [here]( https://shopify.dev/docs/apps/checkout/advanced-checkout-branding#step-4-make-customization-changes).

The results - `GraphqlQueryError: Field 'checkoutBrandingUpsert' doesn't exist on type 'Mutation'`
It fails because I can not figure out how to update the API Version to `2023-10`
I have tried changing & adding types here - node_modules/@shopify/shopify-api/lib/types.d.ts 
I added `October23 = 2023-10`  @ line 13. 
I have tried to hard code it here - `.graphqlrc.js` - remove the `apiVersion` variable and type in `2023-10` 
I added `LATEST_API_VERSION` to my .env just to see if that grabbed it, it did not. 
I tried to look through the CLI, but did not see anything there. 
I tried the ShopifyPartners/All Apps/Api Access - but did not see anything about versions there.

My demo store is under the shopify+ account, I have access to the cart editor and the store is on `Non-transferable
[Checkout Extensibility]preview`.
 
On every try, the headers returned in the log are 
```
........ // info in brackets removed.
emix │     'X-Download-Options': [ '' ],
12:55:48 │ remix │     'X-Frame-Options': [ '' ],
12:55:48 │ remix │     'X-Permitted-Cross-Domain-Policies': [ '' ],
12:55:48 │ remix │     'X-Request-Id': [ '' ],
12:55:48 │ remix │     'X-Shardid': [ '' ],
12:55:48 │ remix │     'X-Shopid': [ '' ],
12:55:48 │ remix │     'X-Shopify-Api-Version': [ '2023-07' ], < ------- this will not change???
12:55:48 │ remix │     'X-Shopify-Stage': [ '' ],
12:55:48 │ remix │     'X-Sorting-Hat-Podid': [ '' ],
12:55:48 │ remix │     'X-Sorting-Hat-Shopid': [ '' ],
12:55:48 │ remix │     'X-Stats-Apiclientid': [ '' ],
12:55:48 │ remix │     'X-Stats-Apipermissionid': [ '' ],
12:55:48 │ remix │     'X-Stats-Userid': [ '' ],
......
```

- `@shopify/shopify-app-remix` version: 1.3.0
- Node version: 18.18.0
- Operating system: Mac OS 12.6.3

## Expected behavior
If I hard code the API version in the `.graphqlrc` file,  I would assume it would use that?
If I use a different version, like `Unstable` it should use it, as that is in the types file. 

Basically, how do I change what version of the Admin API the remix template is using? 

## Actual behavior

Will not change API Version

## Steps to reproduce the problem

1. Create app
2. Follow this https://shopify.dev/docs/apps/checkout/advanced-checkout-branding#step-4-make-customization-changes
3. Change checkoutBranding details using shopify remix app 
4. Changes take effect and API gives me a response. 

Sorry if this is more of question than an issue, but I can not seem to locate how to do this? I assume I am missing something easy somewhere! Thank you!
